app/src/main/java/com/google/android/apps/youtube/app/settings/videoquality/VideoQualitySettingsActivity.java

```diff
@@ -0,0 +1,118 @@
+package com.google.android.apps.youtube.app.settings.videoquality;
+
+import android.annotation.SuppressLint;
+import android.app.Activity;
+import android.content.Context;
+import android.os.Bundle;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageButton;
+import android.widget.TextView;
+
+import app.revanced.integrations.settingsmenu.ReVancedSettingsFragment;
+import app.revanced.integrations.settingsmenu.ReturnYouTubeDislikeSettingsFragment;
+import app.revanced.integrations.settingsmenu.SponsorBlockSettingsFragment;
+import app.revanced.integrations.utils.LogHelper;
+import app.revanced.integrations.utils.ReVancedUtils;
+import app.revanced.integrations.utils.ThemeHelper;
+
+// Hook a dummy Activity to make the Back button work
+public class VideoQualitySettingsActivity extends Activity {
+
+    @Override // android.app.Activity
+    protected void onCreate(Bundle bundle) {
+        final var whiteTheme = "Theme.YouTube.Settings";
+        final var darkTheme = "Theme.YouTube.Settings.Dark";
+
+        final var theme = ThemeHelper.isDarkTheme() ? darkTheme : whiteTheme;
+
+        LogHelper.printDebug(() -> "Using theme: " + theme);
+        setTheme(getIdentifier(theme, "style"));
+
+        super.onCreate(bundle);
+        setContentView(getIdentifier("revanced_settings_with_toolbar", "layout"));
+        initImageButton();
+
+        int fragment = getIdentifier("revanced_settings_fragments", "id");
+
+        String dataString = getIntent().getDataString();
+        if (dataString.equalsIgnoreCase("sponsorblock_settings")) {
+            trySetTitle(getIdentifier("sb_settings", "string"));
+            getFragmentManager()
+                    .beginTransaction()
+                    .replace(fragment, new SponsorBlockSettingsFragment())
+                    .commit();
+        } else if (dataString.equalsIgnoreCase("ryd_settings")) {
+            trySetTitle(getIdentifier("revanced_ryd_settings_title", "string"));
+            getFragmentManager()
+                    .beginTransaction()
+                    .replace(fragment, new ReturnYouTubeDislikeSettingsFragment())
+                    .commit();
+        } else {
+            trySetTitle(getIdentifier("revanced_settings", "string"));
+            getFragmentManager()
+                    .beginTransaction()
+                    .replace(fragment, new ReVancedSettingsFragment())
+                    .commit();
+        }
```

Comment:
This introduces a lot of duplicate code. Please revert it to what it was before.